src/Interpreters/convertFieldToType.cpp

```diff
@@ -572,6 +572,16 @@ Field convertFieldToTypeImpl(const Field & src, const IDataType & type, const ID
 
 }
 
+Field tryConvertFieldToType(const Field & from_value, const IDataType & to_type, const IDataType * from_type_hint, const FormatSettings & format_settings)
+{
+    try
```

Comment:
```suggestion
    /// TODO: implement proper tryConvertFieldToType without try/catch by adding template flag to convertFieldToTypeImpl to not throw an exception.
    try
```

Reply:
`try/catch` is ok for now, but in future better to implement a proper `tryConvertFieldToType` that will avoid throwing and catchin an exception